app/controllers/nonsipp/shares/SharesListController.scala

```diff
@@ -0,0 +1,241 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.nonsipp.shares
+
+import cats.implicits.{catsSyntaxApplicativeId, toShow, toTraverseOps}
+import com.google.inject.Inject
+import config.Constants
+import config.Refined.Max5000
+import controllers.PSRController
+import controllers.actions.IdentifyAndRequireData
+import controllers.nonsipp.shares.SharesListController._
+import eu.timepit.refined.refineMV
+import forms.YesNoPageFormProvider
+import models.SchemeId.Srn
+import models.requests.DataRequest
+import models.{Mode, Pagination, SchemeHoldShare, TypeOfShares}
+import navigation.Navigator
+import pages.nonsipp.shares._
+import play.api.data.Form
+import play.api.i18n.MessagesApi
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents, Result}
+import services.{PsrSubmissionService, SaveService}
+import utils.DateTimeUtils.localDateShow
+import utils.ListUtils._
+import viewmodels.DisplayMessage.{LinkMessage, Message, ParagraphMessage}
+import viewmodels.implicits._
+import viewmodels.models._
+import views.html.TwoColumnsTripleAction
+
+import java.time.LocalDate
+import javax.inject.Named
+import scala.concurrent.{ExecutionContext, Future}
+
+class SharesListController @Inject()(
+  override val messagesApi: MessagesApi,
+  @Named("non-sipp") navigator: Navigator,
+  identifyAndRequireData: IdentifyAndRequireData,
+  val controllerComponents: MessagesControllerComponents,
+  view: TwoColumnsTripleAction,
+  formProvider: YesNoPageFormProvider,
+  psrSubmissionService: PsrSubmissionService,
+  saveService: SaveService
+)(implicit ec: ExecutionContext)
+    extends PSRController {
+
+  val form: Form[Boolean] = SharesListController.form(formProvider)
+
+  def onPageLoad(srn: Srn, page: Int, mode: Mode): Action[AnyContent] = identifyAndRequireData(srn) {
+    implicit request =>
+      // remove once last page is developed
```

Comment:
Page 26r is now done, is this the last page you meant @westwater ?

Reply:
Yeah I think so, I'll raise another PR after this to quickly stitch everything together (don't want to wait too long on this PR incase of another conflict)